docs/spec-files/threading.md

```diff
@@ -0,0 +1,66 @@
+## Threading
+
+DIDComm provides a mechanism for referencing individual messages, the sets of messages that constitute a logically independent interaction (a "run" or "instance" of a protocol), or even the hierarchies of messages that model nested and recursive workflows. This helps with troubleshooting, debugging, error reporting, resend logic, interactions across more than two parties, and patterns of communication that are more flexible or asynchronous than simple request+response. It is the foundation for extremely powerful protocol features.
+
+### Message IDs
+
+All plaintext DIDComm messages MUST have an `id` property, declared in the JWM header. A message without an `id` property SHOULD be considered invalid and SHOULD be rejected; it MUST NOT be interpreted as part of a multi-message interaction.
```

Comment:
If we have a one-way, one-message protocol (e.g., something that sends notifications only), I don't think that threading is needed. Therefore, I put "SHOULD" in rather than MUST. Changing this to a MUST makes sense for two-way, multi-step interactions, but I don't think it does for one-way.

Reply:
Agreed, this change doesn't make sense now. Let's go with what you have.

Reply:
Should we add a note explaining why it's only "SHOULD"? Or maybe, put in "MUST" for any multi-step interaction?

Reply:
I think it's fine to leave it as just "SHOULD" without explicit mention. Typically when I read documents these days "SHOULD" means to me "It's done this way unless you got a reason to not do it" which relies upon the reader to be able to come up with (and then they know they're making an exception to a soft rule).